src/Autofac/ContainerBuilder.cs

```diff
@@ -98,6 +99,17 @@ internal ContainerBuilder(IDictionary<string, object> properties)
         /// </value>
         public IDictionary<string, object> Properties { get; }
 
+        /// <summary>
+        /// Sets the maximum size of the stack used when resolving a single component (and all of its transitive dependencies) from the container.
+        /// If this limit is exceeded the component will be treated as a circular dependency.
+        /// </summary>
+        /// <param name="maximumStackDepth">Maximum size of the stack used for resolve operations.</param>
+        public ContainerBuilder WithMaxResolveStackDepth(int maximumStackDepth)
```

Comment:
- Could this be an extension method instead of something changing the `ContainerBuilder` public API?
- We need logic to ensure folks don't set this to zero, a negative number, or something that will shoot themselves in the foot. Perhaps the minimum is the current default of 50?